jax/_src/api.py

```diff
@@ -333,15 +335,67 @@ def jit(
     DeviceArray([   0,    1,  256, 6561], dtype=int32)
   """
   if FLAGS.experimental_cpp_jit and not config.jax_dynamic_shapes:
-    return _cpp_jit(fun, static_argnums, static_argnames, device, backend,
+    return _jit("cpp", fun, static_argnums, static_argnames, device, backend,
                     donate_argnums, inline, keep_unused)
   else:
-    return _python_jit(fun, static_argnums, static_argnames, device, backend,
+    return _jit("python", fun, static_argnums, static_argnames, device, backend,
                        donate_argnums, inline, keep_unused, abstracted_axes)
 
+def _jit(
```

Comment:
This refactoring is done to reduce code duplication in `_jit_cpp` and `_jit_python`.

This does lead to some small changes in tests